src/interface/report/Results/PhaseSelector.tsx

```diff
@@ -36,13 +47,19 @@ class PhaseSelector extends React.PureComponent {
   //builds a dictionary of phases / phase instances to keep track of in order to be able to attribute a unique "key" to each phase for the dropdown
   //without losing the actual key (and without having to for example replace an "instance token" like an underscore)
   buildPhases() {
-    const phases = [];
+    const phases : {
+      name: string; 
+      key: string;
+      instance: number;
+      start: number;
+      multiple?: boolean;
+    }[] = [];
     Object.keys(this.props.phases).forEach(key => {
       const phase = this.props.phases[key];
-      if (phase.start.length !== phase.end.length) {
-        phases.push({ name: phase.name, key: key, start: phases.start[0], instance: 0 });
+      if (phase!.start!.length !== phase!.end!.length) {
```

Comment:
```suggestion
      if (phase.start!.length !== phase.end!.length) {
```

Reply:
Needing so many ! is a sign that types can be improved. Is it maybe an idea to have two different types for Phases; e.g. `PhaseConfig` and `Phase`? I'm not sure how it's used so I can't really tell

Reply:
wasn't entirely sure how to solve this since the phase type is probably too overloaded for its own good but I think I have a solution for this in the new commits? def need more feedback on that one tho